specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,959 @@
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// Options for unpacking a package
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass UnbundleOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into UnpackPackage
+        /// </summary>
+        UnbundleOptions();
+
+        /// <summary>
+        /// The output folder to unpack the bundle package into.
+        /// </summary>
+        String UnbundledPackageRootDirectory{ get; set; };
+        /// <summary>
+        /// If true, the operation will overwrite existing files in the UnbundledPackageRootDirectory
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// 
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass CreateMountableImageOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into CreateMountableImage
+        /// </summary>
+        CreateMountableImageOptions();
+
+        /// <summary>
+        /// Output path for the image file.
+        /// Valid file extensions are vhd, vhdx, and CIM.
+        /// </summary>
+        String ImageFilePath{ get; set; };
+
+        /// <summary>
+        /// Root directory path for packages inside the image.
+        /// Defaults to "WindowsApps"
+        /// </summary>
+        String PackageRootDirectoryInImage{ get; set; };
+
+        /// <summary>
+        /// If true, the operation will overwrite an existing file in the ImageFilePath
+        /// when unpacking.
+        /// If false, the operation will fail if the file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// Static methods for creating and unpacking packages.
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass MakeMSIXManager
+    {
+        /// <summary>
+        /// Creates packages from the directoryPathToPack as specified by the packOptions.
+        /// </summary>
+        static Windows.Foundation.IAsyncAction Pack(String directoryPathToPack, PackOptions packOptions);
+        /// <summary>
+        /// Creates bundles from the directoryPathToBundle as specified by the bundleOptions.
+        /// </summary>
+        static Windows.Foundation.IAsyncAction Bundle(String directoryPathToBundle, BundleOptions bundleOptions);
+        /// <summary>
+        /// Unpacks a package at packageFilePathToUnpack as specified by the unpackOptions.
+        /// </summary>
+        static Windows.Foundation.IAsyncAction Unpack(String packageFilePathToUnpack, UnpackOptions unpackOptions);
```

Comment:
All these take 1 required input parameter and 1 required Options object but you don't have to set any of these properties in the Options objects. Is that distinction useful?

What's the minimum required parameters for these APIs? It's hard to imagine just 1 input filename is required, but no output filename? What's required to use these e.g. Pack = inputPath + outputFileName(OrPath if LooseFile)

Implies these APIs should take 3 parameters: input, output, options

Reply:
Good point, I like that a lot better and will change. I think this is probably a remnant of having earlier versions of this api where things were grouped together more and the minimal set of options was less easy to discover. Agree that making that clear is a primary concern.